### PR TITLE
Export type aliases for `Storage` and `Footprint`; update wasm files

### PR DESCRIPTION
**Pull request by jayz22**
### What

Export type aliases for `Storage` and `Footprint`.
Update the wasm test files.

### Why

Make life easier. These types are repeatedly copied in the `stellar-core` rust code base (and possible in other crates like `soroban-cli`). 

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: soroban-env-host/src/storage.rs [hunk 1]
```diff
@@ -17,6 +17,9 @@ use crate::xdr::{LedgerEntry, LedgerKey, ScHostStorageErrorCode};
 use crate::Host;
 use crate::{host::metered_map::MeteredOrdMap, HostError};
 
+pub type FootprintType = MeteredOrdMap<Rc<LedgerKey>, AccessType, Budget>;
+pub type StorageMapType = MeteredOrdMap<Rc<LedgerKey>, Option<Rc<LedgerEntry>>, Budget>;
```
Comment (leighmcculloch) [inline soroban-env-host/src/storage.rs:21]: Suggest naming these `FootprintMap and `StorageMap`, since both are maps.

**Review by leighmcculloch — APPROVED** [review]
One suggestion, but happy to merge without it, lmk if you want merge.

File: soroban-env-host/src/storage.rs (hunk 1 above)
Comment (jayz22) [reply soroban-env-host/src/storage.rs]: Done.

**Comment by jayz22** [comment]
> One suggestion, but happy to merge without it, lmk if you want merge.

Thanks, ready to go.

**Review by sisuresh — APPROVED** [review]

**Pull request merged**